homeassistant/components/tado/climate.py

```diff
@@ -289,7 +289,15 @@ def preset_modes(self):
 
     def set_preset_mode(self, preset_mode):
         """Set new preset mode."""
-        pass
+        print(preset_mode)
+        if preset_mode is None:
+            return
+        elif preset_mode == "away":
+            _LOGGER.debug("Setting presence to away")
+            self._tado.set_presence("AWAY")
+        else:
```

Comment:
Please use https://github.com/home-assistant/core/blob/992daa4a44f20664423d0a405f021b51d5057726/homeassistant/components/climate/const.py#L51
here to filter out other `preset_modes`